pkgs/test/default.nix

```
@@ -24,6 +24,11 @@ with pkgs;
 
   config = callPackage ./config.nix { };
 
+  # we can't add 'nixpkgs.curl.tests' to hydra jobs due to 'passthru' being stripped
+  passthruTestsForChannelBlockers = recurseIntoAttrs {
```

```suggestion
  # we can't add 'nixpkgs.curl.tests' to hydra jobs due to 'tests' (and 'passthru') being stripped
  packageTestsForChannelBlockers = recurseIntoAttrs {
```

The fact that they're on a package is more significant than `passthru`, which is just an implementation detail of `mkDerivation`.

`passthru` was introduced in https://github.com/NixOS/nixpkgs/commit/1014ca2ad55761b54366d853da45a11194d4f531 (2006) and only exposed the attrs inside of it. The behavior `passthru.passthru = passthru` was only added later for inspection https://github.com/NixOS/nixpkgs/commit/84fba68006259f1732a9580ba2bdcbc300300c89:

> to distinguish after the fact which attributes were part of the derivation and which came from passthru.

However, the root cause of any such ambiguity comes from the mistake that `mkDerivation` exposes all its argument attributes in the returned package attrset.

Since RFC 92, not all packages will be defined through `mkDerivation` anymore, so we should try not to perpetuate its jargon and keep things simple.